api/v1/controllers/RSVPController.js

```diff
@@ -0,0 +1,146 @@
+        })
+        .catch(function(error) {
+            return next(error);
+        });
+}
+
+function fetchRSVPByUser(req, res, next) {
+    services.RegistrationService
+        .findAttendeeByUser(req.user)
+        .then(function(attendee) {
+            services.RSVPService
+                .findRSVPByAttendee(attendee)
+                .then(function(rsvp){
+                    res.body = rsvp.toJSON();
+
+                    return next();
+                })
+        })
+        .catch(function(error) {
+            return next(error);
+        })
+}
+
+function fetchRSVPByAttendeeId(req, res, next) {
+    services.RegistrationService
+        .findAttendeeById(req.params.id)
+        .then(function(attendee) {
+            services.RSVPService
+                .findRSVPByAttendee(attendee)
+                .then(function(rsvp){
+                    res.body = rsvp.toJSON();
+
+                    return next();
+                })
+        })
+        .catch(function(error) {
+            return next(error);
+        })
+}
+
+function updateRSVPByUser(req, res, next) {
+    services.RegistrationService
+        .findAttendeeByUser(req.user)
+        .then(function(attendee) {
+            services.RSVPService
+                .findRSVPByAttendee(attendee)
+                .then(function (rsvp) {
+                    if(_addToList(rsvp, req.body))
+                        services.MailService.addToList(req.user, mail.lists.attendees);
+                    if(_removeFromList(rsvp, req.body))
+                        services.MailService.removeFromList(req.user, mail.lists.attendees);
+
+                    return services.RSVPService.updateRSVP(rsvp, req.body);
+                })
+                .then(function(rsvp){
+                    res.body = rsvp.toJSON();
+
+                    return next();
+                })
+        })
+        .catch(function (error) {
+            return next(error);
+        });
+}
+
+function updateRSVPById(req, res, next) {
+    services.RegistrationService
+        .findAttendeeById(req.params.id)
+        .then(function(attendee) {
+            services.RSVPService
+                .findRSVPByAttendee(attendee)
+                .then(function (rsvp) {
+                    if(_addToList(rsvp, req.body))
+                        services.MailService.addToList(req.user, mail.lists.attendees);
+                    if(_removeFromList(rsvp, req.body))
+                        services.MailService.removeFromList(req.user, mail.lists.attendees);
+
+                    return services.RSVPService.updateRSVP(rsvp, req.body);
+                })
+                .then(function(rsvp){
+                    res.body = rsvp.toJSON();
+
+                    return next();
+                })
+        })
+        .catch(function (error) {
+            return next(error);
+        });
+}
+
+router.use(bodyParser.json());
+router.use(middleware.auth);
+
+router.post('/', middleware.request(requests.RSVPRequest),
```

Comment:
Since this only accounts for attendee registration, it should probably be a route like `/attendee`. Otherwise, if we choose to do RSVPs through this system for mentors later, we'll have to move things around.